src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,32 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int LOGIN_MIN_LENGTH = 6;
+    private static final int PASSWORD_MIN_LENGTH = 6;
+    private static final int USER_MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("Sorry but someone steals your name. "
+                    + "Go to change your passport");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < LOGIN_MIN_LENGTH) {
```

Comment:
why did you check login for null after you got by login from storageDao?

Reply:
cause I'm checking for existing this login in storage
that's how it was before:

User newUser = storageDao.get(user.getLogin());
        if (newUser != null) {
        ...
        }